toolkit/offchain/src/csl.rs

```diff
@@ -150,6 +166,38 @@ pub(crate) fn get_first_validator_budget(
 	Ok(convert_ex_units(&validator_budget.budget))
 }
 
+#[derive(Debug, Clone)]
+pub struct ScriptExUnits {
+	pub mint_ex_units: Vec<ExUnits>,
+	pub spend_ex_units: Vec<ExUnits>,
+}
+
+impl ScriptExUnits {
+	fn new() -> Self {
+		ScriptExUnits { mint_ex_units: Vec::new(), spend_ex_units: Vec::new() }
+	}
+}
+
+pub(crate) fn get_validator_budgets(
+	responses: Vec<OgmiosEvaluateTransactionResponse>,
+) -> Result<ScriptExUnits, JsError> {
+	let mut ex_units = ScriptExUnits::new();
+	let mut mint_ex_units = vec![];
+	let mut spend_ex_units = vec![];
+	for response in responses.iter() {
+		if response.validator.purpose == "mint" {
+			mint_ex_units.push((convert_ex_units(&response.budget), response.validator.index));
+		} else {
+			spend_ex_units.push((convert_ex_units(&response.budget), response.validator.index));
+		}
+	}
+	mint_ex_units.sort_by(|a, b| a.1.cmp(&b.1));
```

Comment:
Please add comment how are they sorted. Took me a while to understand that is is by index.

Reply:
I reworked it instead to avoid the mutbility etc.